include/flash_pm.h

```diff
@@ -0,0 +1,32 @@
+#ifndef ZEPHYR_INCLUDE_FLASH_PM_H_
+#define ZEPHYR_INCLUDE_FLASH_PM_H_
+
+/* Default DT node with fixed-partitions compatibility set */
+#define DEFAULT_DT_FIXED_PARTITIONS DT_INST(0, fixed_partitions)
+
+/* Right now only DT definitions are supported, and required */
+#if DT_NODE_HAS_COMPAT(DEFAULT_DT_FIXED_PARTITIONS, fixed_partitions)
+#define FLASH_AREA_LIST DT_CHILD_NODELABELS(DEFAULT_DT_FIXED_PARTITIONS)
+#else
+#error "Expected default fixed partitions table"
+#endif
+
+/* Number of flash areas */
+#define FLASH_AREA_N UTIL_LIST_LEN(FLASH_AREA_LIST)
+
+#define FLASH_AREA_ID(lbl) __dt_flash_defined_partition__##lbl
+
+/* Generate sequential identifiers for flash areas. */
+#define _FLASH_AREA_ID_GEN(lbl) FLASH_AREA_ID(lbl),
+enum __dt_flash_defined_partitions {
+	UTIL_FOR_LIST(_FLASH_AREA_ID_GEN, FLASH_AREA_LIST)
+};
+
+#define FLASH_AREA_OFFSET(lbl)	DT_REG_ADDR(DT_NODELABEL(lbl))
+#define FLASH_AREA_SIZE(lbl)	DT_REG_SIZE(DT_NODELABEL(lbl))
+#define FLASH_AREA_PTABLE(lbl)	DT_PARENT(DT_NODELABEL(lbl))
```

Comment:
Can you explain the need for these macros to be in the API header instead of the default implementation file?

Reply:
They are supposed to be provided for a programmer to access FLASH_AREA information.